src/main/java/de/zalando/aruha/nakadi/controller/EventTypeController.java

```diff
@@ -154,12 +157,23 @@ public EventTypeController(final EventTypeRepository eventTypeRepository, final
         }
     }
 
-    private void validatePartitioningStrategy(final EventType eventType) throws NoSuchPartitioningStrategyException {
+    private void validatePartitioningStrategy(final EventType eventType) throws NoSuchPartitioningStrategyException,
+            InvalidEventTypeException {
         final PartitionResolutionStrategy partitioningStrategy = eventType.getPartitionResolutionStrategy();
-        if (partitioningStrategy != null && !partitionResolver.strategyExists(partitioningStrategy.getName())) {
+        if (!partitionResolver.strategyExists(partitioningStrategy.getName())) {
             throw new NoSuchPartitioningStrategyException("partitioning strategy does not exist: " +
                     partitioningStrategy.getName());
         }
+        else if (HASH_STRATEGY.equals(partitioningStrategy.getName()) &&
+                eventType.getPartitioningKeyFields().isEmpty()) {
+            throw new InvalidEventTypeException("partitioning_key_fields field should be set for " +
+                    "partition resolution strategy 'hash'");
+        }
+        else if (USER_DEFINED_STRATEGY.equals(partitioningStrategy.getName()) &&
+                UNDEFINED.equals(eventType.getCategory())) {
```

Comment:
Why is it forbidden for `undefined` category? We could treat the `metadata` field as a reserved one (even for `undefined`) and add it silently to the effective schema. Surely, if this is the case, event type registration should fail if the schema contains a `metadata` field.